scripts/west_commands/runners/linkserver.py

```diff
@@ -0,0 +1,236 @@
+                            help='interface to use (index, no serial number), default is 1')
+
+        parser.add_argument('--tui', default=False, action='store_true',
+                            help='if given, GDB uses -tui')
+
+        parser.add_argument('--gdb-port', default=DEFAULT_LINKSERVER_GDB_PORT,
+                            help='gdb port to open, defaults to {}'.format(
+                               DEFAULT_LINKSERVER_GDB_PORT))
+
+        parser.add_argument('--semihost-port', default=DEFAULT_LINKSERVER_SEMIHOST_PORT,
+                            help='semihost port to open, defaults to the empty string '
+                            'and runs a gdb server')
+        # keep this, we have to assume that the default 'commander' is on PATH
+        parser.add_argument('--linkserver', default=DEFAULT_LINKSERVER_EXE,
+                            help=f'''LinkServer executable, default is
+                            {DEFAULT_LINKSERVER_EXE}''')
+        # user may need to override settings.
+        parser.add_argument('--override', required=False, action='append',
+                            help=f'''configuration overrides as defined bylinkserver. Example: /device/memory/0/location=0xcafecafe''')
+
+    @classmethod
+    def do_create(cls, cfg, args):
+        return LinkServerBinaryRunner(cfg, args.device,
+                                 linkserver=args.linkserver,
+                                 dt_flash=args.dt_flash,
+                                 erase=args.erase,
+                                 probe=args.probe,
+                                 semihost_port=args.semihost_port,
+                                 gdb_port=args.gdb_port,
+                                 override=args.override,
+                                 tui=args.tui, tool_opt=args.tool_opt)
+
+## UPDATE
+    def print_gdbserver_message(self):
+        if not self.thread_info_enabled:
+            thread_msg = '; no thread info available'
+        elif self.supports_thread_info:
+            thread_msg = '; thread info enabled'
+        else:
+            thread_msg = '; update J-Link software for thread info'
+        self.logger.info('J-Link GDB server running on port '
+                         f'{self.gdb_port}{thread_msg}')
+
+    @property
+    def linkserver_version(self):
+
+        if not hasattr(self, '_linkserver_version'):
+            linkserver_version_cmd=[self.linkserver, "-v"]
+            ls_output=self.check_output(linkserver_version_cmd)
+            self._linkserver_version = str(ls_output.split()[1].decode())
+
+        return self._linkserver_version
+
+    @property
+    def linkserver_version_str(self):
+
+        return self.linkserver_version
+
+
+    @property
+    def supports_thread_info(self):
+        # TODO: check if linkserver needs this??
+        return True
+
+    def do_run(self, command, **kwargs):
+
+        # Convert commander to a real absolute path. We need this to
```

Comment:
Are you sure this is necessary? In my experience, shutil.which() (which underlies self.require()) returns absolute paths. Did you have a different experience?